// darth-orga-ui/publish/mock/db.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

var _this = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

// 用mockjs模拟生成数据
var Mock = require('mockjs');

var mr = Mock.Random; //提取mock的随机对象
//随机id和图片

var mapData = function mapData(n) {
  (0, _newArrowCheck2["default"])(this, _this);
  var data = [];

  for (var i = 1; i <= n; i++) {
    data.push({
      id: i,
      title: "@ctitle(8,12)",
      des: "@csentence(10, 20)",
      time: "@integer(1553425967486,1553475967486)",
      detail: {
        auth: "@cname()",
        content: "@cparagraph(10,40)",
        auth_icon: mr.image('50x50', mr.color(), mr.cword(1))
      }
    });
  }

  return data;
}.bind(void 0); //json-server 要对象||函数(返回mock后的数据)


module.exports = _objectSpread({}, Mock.mock({
  'home': mapData(32),
  //解决 auth_icon 不随机
  'follow': mapData(21),
  'column': mapData(11),
  'banner|2': [{
    // 属性 id 是一个自增数，起始值为 1，每次增 1
    'id|+1': 1,
    title: "@ctitle(4,8)",
    //标题型中文4到8个字
    sub_title: "@ctitle(6,12)",
    banner: mr.image('750x501', mr.color(), mr.cword(4, 10)),
    //banner不变
    time: "@integer(1553425967486,1553475967486)",
    detail: {
      icon: mr.image('20x20', mr.color(), mr.cword(1, 2)),
      //20X20尺寸
      auth: "@cname()",
      //百家姓
      content: "@cparagraph(10,40)" //正文

    }
  }]
}));
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(mr, "mr", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/mock/db.js");
  reactHotLoader.register(mapData, "mapData", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/mock/db.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
