// darth-orga/publish/common/language/i18n.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _i18next = _interopRequireDefault(require("i18next"));

var _i18nextBrowserLanguagedetector = _interopRequireDefault(require("i18next-browser-languagedetector"));

var _i18nextHttpBackend = _interopRequireDefault(require("i18next-http-backend"));

var _reactI18next = require("react-i18next");

var _translation = _interopRequireDefault(require("./cn/translation.json"));

var _translation2 = _interopRequireDefault(require("./en/translation.json"));

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

_i18next["default"].use(_i18nextHttpBackend["default"]).use(_i18nextBrowserLanguagedetector["default"]) //嗅探当前浏览器语言
.use(_reactI18next.initReactI18next) //init i18next
.init({
  //引入资源文件
  resources: {
    zh: {
      translation: _translation["default"]
    },
    en: {
      translation: _translation2["default"]
    }
  },
  //选择默认语言，选择内容为上述配置中的key，即en/zh
  fallbackLng: "zh",
  lng: 'zh',
  debug: false,
  interpolation: {
    escapeValue: false // not needed for react as it escapes by default

  }
});

var _default = _i18next["default"];
var _default2 = _default;
exports["default"] = _default2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(_default, "default", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/common/language/i18n.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
