// darth-orga/publish/modules/user/containers/user.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

require("antd/es/table/style/css");

var _table = _interopRequireDefault(require("antd/es/table"));

require("antd/es/row/style/css");

var _row = _interopRequireDefault(require("antd/es/row"));

require("antd/es/button/style/css");

var _button = _interopRequireDefault(require("antd/es/button"));

require("antd/es/col/style/css");

var _col = _interopRequireDefault(require("antd/es/col"));

require("antd/es/input/style/css");

var _input = _interopRequireDefault(require("antd/es/input"));

require("antd/es/space/style/css");

var _space = _interopRequireDefault(require("antd/es/space"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

require("antd/es/tabs/style/css");

var _tabs = _interopRequireDefault(require("antd/es/tabs"));

var _react = _interopRequireWildcard(require("react"));

var _mobxReact = require("mobx-react");

require("./user.scss");

var _breadCrumb = _interopRequireDefault(require("../../../common/bread-crumb"));

var _addUser = _interopRequireDefault(require("../components/add-user"));

var _utils = require("../../../common/utils");

var _this = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var TabPane = _tabs["default"].TabPane;

var DarthUser = __signature__(function DarthUser(props) {
  var _this2 = this;

  (0, _newArrowCheck2["default"])(this, _this);
  var darthUseRStore = props.darthUseRStore;
  var userData = darthUseRStore.userData,
      getUserList = darthUseRStore.getUserList,
      editUser = darthUseRStore.editUser,
      addUser = darthUseRStore.addUser,
      deleteUser = darthUseRStore.deleteUser;

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      visible = _useState2[0],
      setVisible = _useState2[1];

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      isEdit = _useState4[0],
      setEdit = _useState4[1];

  var _useState5 = (0, _react.useState)({}),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      editData = _useState6[0],
      setEditData = _useState6[1];

  (0, _react.useEffect)(function () {
    (0, _newArrowCheck2["default"])(this, _this2);
    getUser();
  }.bind(this), []);

  var getUser = function getUser() {
    var _this3 = this;

    (0, _newArrowCheck2["default"])(this, _this2);
    (0, _utils.baseFetch)('post', 'api/user/findAllUser').then(function (res) {
      (0, _newArrowCheck2["default"])(this, _this3);

      if (!res.code) {
        getUserList(res.data);
      }
    }.bind(this));
  }.bind(this);

  var crumbRoutes = [{
    path: 'index',
    breadcrumbName: '用户管理 '
  }, {
    path: 'first',
    breadcrumbName: '用户列表'
  }];
  var columns = [{
    title: '姓名',
    dataIndex: 'name',
    key: 'name'
  }, {
    title: '手机号',
    dataIndex: 'phone',
    key: 'phone'
  }, {
    title: '邮箱',
    dataIndex: 'email',
    key: 'email'
  }, {
    title: '部门',
    dataIndex: 'org',
    key: 'org'
  }, {
    title: '类别',
    dataIndex: 'userType',
    key: 'userType',
    render: function render(text) {
      (0, _newArrowCheck2["default"])(this, _this2);

      switch (text) {
        case 2:
          return '第三方';

        case 1:
          return '内部';

        default:
          return '--';
      }
    }.bind(this)
  }, {
    title: '操作',
    key: 'action',
    render: function render(text, record) {
      var _this4 = this;

      (0, _newArrowCheck2["default"])(this, _this2);
      return /*#__PURE__*/_react["default"].createElement(_space["default"], {
        size: "middle"
      }, /*#__PURE__*/_react["default"].createElement("a", {
        onClick: function () {
          (0, _newArrowCheck2["default"])(this, _this4);
          return onEditUser(record);
        }.bind(this)
      }, "\u7F16\u8F91"), /*#__PURE__*/_react["default"].createElement("a", {
        onClick: function () {
          (0, _newArrowCheck2["default"])(this, _this4);
          return onDeleteUser(record.id);
        }.bind(this)
      }, "\u5220\u9664"));
    }.bind(this)
  }];

  var onDeleteUser = function onDeleteUser(id) {
    var _this5 = this;

    (0, _newArrowCheck2["default"])(this, _this2);
    (0, _utils.baseFetch)('post', 'api/user/deleteUser', {
      id: id
    }).then(function (res) {
      (0, _newArrowCheck2["default"])(this, _this5);

      if (!res.code) {
        deleteUser(id);
      }
    }.bind(this));
  }.bind(this);

  var onEditUser = function onEditUser(data) {
    var _this6 = this;

    (0, _newArrowCheck2["default"])(this, _this2);
    var formData = new FormData();
    formData.append('id', data.id);
    (0, _utils.baseFetch)('post', 'api/user/findUser', formData).then(function (res) {
      (0, _newArrowCheck2["default"])(this, _this6);

      if (!res.code) {
        setEditData(res.data);
        setVisible(true);
        setEdit(true);
      }
    }.bind(this));
  }.bind(this);

  var changTab = function changTab(key) {
    (0, _newArrowCheck2["default"])(this, _this2);
    console.log(key);
  }.bind(this);

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "darth-user"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "darth-user-detail"
  }, /*#__PURE__*/_react["default"].createElement(_breadCrumb["default"], {
    routes: crumbRoutes
  }), /*#__PURE__*/_react["default"].createElement(_row["default"], null, /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 8
  }, /*#__PURE__*/_react["default"].createElement(_input["default"], {
    placeholder: '搜索用户名、帐号'
  })), /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 4,
    offset: 12
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: 'darth-user-detail-buttons'
  }, /*#__PURE__*/_react["default"].createElement(_button["default"], {
    onClick: function () {
      (0, _newArrowCheck2["default"])(this, _this2);
      return setVisible(true);
    }.bind(this)
  }, "+\u6DFB\u52A0\u7528\u6237"), /*#__PURE__*/_react["default"].createElement(_button["default"], null, "\u5BFC\u5165\u7528\u6237")))), /*#__PURE__*/_react["default"].createElement(_row["default"], null, /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 24
  }, /*#__PURE__*/_react["default"].createElement(_tabs["default"], {
    onChange: changTab,
    type: "card"
  }, /*#__PURE__*/_react["default"].createElement(TabPane, {
    tab: "\u5168\u90E8",
    key: "1"
  }, /*#__PURE__*/_react["default"].createElement(_table["default"], {
    bordered: true,
    columns: columns,
    dataSource: userData,
    rowKey: function (record) {
      (0, _newArrowCheck2["default"])(this, _this2);
      return record.name + record.id;
    }.bind(this)
  })), /*#__PURE__*/_react["default"].createElement(TabPane, {
    tab: "\u5185\u90E8",
    key: "2"
  }, "Content of Tab Pane 2"), /*#__PURE__*/_react["default"].createElement(TabPane, {
    tab: "\u5916\u90E8",
    key: "3"
  }, "Content of Tab Pane 3")), ",")), /*#__PURE__*/_react["default"].createElement(_addUser["default"], (0, _extends2["default"])({}, props, {
    visible: visible,
    onCancel: function () {
      (0, _newArrowCheck2["default"])(this, _this2);
      return setVisible(false);
    }.bind(this),
    addUser: addUser,
    isEdit: isEdit,
    data: editData,
    editUser: editUser,
    getList: getUser
  }))));
}, "useState{[visible, setVisible](false)}\nuseState{[isEdit, setEdit](false)}\nuseState{[editData, setEditData]({})}\nuseEffect{}").bind(void 0);

var _default = (0, _mobxReact.inject)('darthUseRStore')((0, _mobxReact.observer)(DarthUser));

var _default2 = _default;
exports["default"] = _default2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(TabPane, "TabPane", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/modules/user/containers/user.js");
  reactHotLoader.register(DarthUser, "DarthUser", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/modules/user/containers/user.js");
  reactHotLoader.register(_default, "default", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/modules/user/containers/user.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
