// darth-orga/publish/common/bread-crumb/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("antd/es/breadcrumb/style/css");

var _breadcrumb = _interopRequireDefault(require("antd/es/breadcrumb"));

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

var _react = _interopRequireDefault(require("react"));

var _reactRouterDom = require("react-router-dom");

require("./index.scss");

var _this = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var OrgaBreadCrumb = function OrgaBreadCrumb(props) {
  var _this2 = this;

  (0, _newArrowCheck2["default"])(this, _this);
  var routes = props.routes;

  var itemRender = function itemRender(route, params, routes, paths) {
    (0, _newArrowCheck2["default"])(this, _this2);
    var last = routes.indexOf(route) === routes.length - 1;
    return last ? /*#__PURE__*/_react["default"].createElement("span", null, route.breadcrumbName) : /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Link, {
      to: paths.join('/')
    }, route.breadcrumbName);
  }.bind(this);

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: 'orga-breadcrumb'
  }, /*#__PURE__*/_react["default"].createElement(_breadcrumb["default"], {
    itemRender: itemRender,
    routes: routes,
    separator: ">"
  }));
}.bind(void 0);

var _default = OrgaBreadCrumb;
var _default2 = _default;
exports["default"] = _default2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(OrgaBreadCrumb, "OrgaBreadCrumb", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/common/bread-crumb/index.js");
  reactHotLoader.register(_default, "default", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/common/bread-crumb/index.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
