// darth-orga-ui/publish/lib/darthUser/components/addUser/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("antd/es/modal/style/css");

var _modal = _interopRequireDefault(require("antd/es/modal"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

require("antd/es/select/style/css");

var _select = _interopRequireDefault(require("antd/es/select"));

require("antd/es/input/style/css");

var _input = _interopRequireDefault(require("antd/es/input"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

require("antd/es/form/style/css");

var _form = _interopRequireDefault(require("antd/es/form"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

var _react = _interopRequireWildcard(require("react"));

var _utils = require("../../../../utils");

require("./inde.scss");

var _this = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var layout = {
  labelCol: {
    span: 6
  },
  wrapperCol: {
    span: 18
  }
};

var AddUser = __signature__(function AddUser(props) {
  var _this2 = this;

  (0, _newArrowCheck2["default"])(this, _this);
  var visible = props.visible,
      onCancel = props.onCancel,
      addUser = props.addUser,
      _props$isEdit = props.isEdit,
      isEdit = _props$isEdit === void 0 ? false : _props$isEdit,
      _props$data = props.data,
      data = _props$data === void 0 ? {} : _props$data,
      editUser = props.editUser;

  var _Form$useForm = _form["default"].useForm(),
      _Form$useForm2 = (0, _slicedToArray2["default"])(_Form$useForm, 1),
      form = _Form$useForm2[0];

  (0, _react.useEffect)(function () {
    (0, _newArrowCheck2["default"])(this, _this2);

    if (isEdit) {
      form.setFieldsValue({
        name: data.name
      });
      form.setFieldsValue({
        phone: data.phone
      });
      form.setFieldsValue({
        email: data.email
      });
      form.setFieldsValue({
        userType: data.userType
      });
    }
  }.bind(this), [isEdit, data]);

  var handleOk = function handleOk() {
    (0, _newArrowCheck2["default"])(this, _this2);
    form.submit();
  }.bind(this);

  var handleCancel = function handleCancel() {
    (0, _newArrowCheck2["default"])(this, _this2);
    form.resetFields();
    onCancel();
  }.bind(this);

  var onFinish = function onFinish(values) {
    var _this3 = this;

    (0, _newArrowCheck2["default"])(this, _this2);

    if (isEdit) {
      (0, _utils.baseFetch)('post', 'api/user/updateUser', _objectSpread(_objectSpread({}, values), {}, {
        id: data.id
      })).then(function (res) {
        (0, _newArrowCheck2["default"])(this, _this3);

        if (!res.code) {
          editUser(_objectSpread(_objectSpread({}, values), {}, {
            id: data.id
          }));
          handleCancel();
        }
      }.bind(this));
    } else {
      (0, _utils.baseFetch)('post', 'api/user/createUser', values).then(function (res) {
        (0, _newArrowCheck2["default"])(this, _this3);

        if (!res.code) {
          addUser(_objectSpread({
            id: res.data
          }, values));
          handleCancel();
        }
      }.bind(this));
    }
  }.bind(this);

  return /*#__PURE__*/_react["default"].createElement(_modal["default"], {
    title: "\u6DFB\u52A0\u7528\u6237",
    visible: visible,
    onOk: handleOk,
    onCancel: handleCancel,
    closable: false
  }, /*#__PURE__*/_react["default"].createElement(_form["default"], (0, _extends2["default"])({}, layout, {
    form: form,
    onFinish: onFinish
  }), /*#__PURE__*/_react["default"].createElement(_form["default"].Item, {
    label: "\u59D3\u540D",
    name: "name",
    rules: [{
      required: true,
      message: '用户名不能包含非法字符，如&,%，&，#……等'
    }]
  }, /*#__PURE__*/_react["default"].createElement(_input["default"], null)), /*#__PURE__*/_react["default"].createElement(_form["default"].Item, {
    label: "\u624B\u673A\u53F7",
    name: "phone",
    rules: [{
      required: true,
      message: '帐号必填'
    }]
  }, /*#__PURE__*/_react["default"].createElement(_input["default"], null)), /*#__PURE__*/_react["default"].createElement(_form["default"].Item, {
    label: "\u90AE\u7BB1",
    name: "email",
    rules: [{
      required: true,
      message: '帐号必填'
    }]
  }, /*#__PURE__*/_react["default"].createElement(_input["default"], null)), /*#__PURE__*/_react["default"].createElement(_form["default"].Item, {
    label: "\u7528\u6237\u7C7B\u578B",
    name: "userType",
    rules: [{
      required: true,
      message: '用户类型必填!'
    }]
  }, /*#__PURE__*/_react["default"].createElement(_select["default"], null, /*#__PURE__*/_react["default"].createElement(_select["default"].Option, {
    key: '1',
    value: 1
  }, "\u5185\u90E8"), /*#__PURE__*/_react["default"].createElement(_select["default"].Option, {
    key: '2',
    value: 2
  }, "\u7B2C\u4E09\u65B9")))));
}, "useForm{[form]}\nuseEffect{}").bind(void 0);

var _default = AddUser;
var _default2 = _default;
exports["default"] = _default2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(layout, "layout", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthUser/components/addUser/index.js");
  reactHotLoader.register(AddUser, "AddUser", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthUser/components/addUser/index.js");
  reactHotLoader.register(_default, "default", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthUser/components/addUser/index.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
