// darth-orga/publish/modules/org/store/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DARTH_ORG_STORE = exports.DarthOrgStore = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

var _initializerDefineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/initializerDefineProperty"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _applyDecoratedDescriptor2 = _interopRequireDefault(require("@babel/runtime/helpers/applyDecoratedDescriptor"));

var _initializerWarningHelper2 = _interopRequireDefault(require("@babel/runtime/helpers/initializerWarningHelper"));

var _mobx = require("mobx");

var _class, _descriptor, _descriptor2, _descriptor3, _descriptor4, _descriptor5, _descriptor6, _descriptor7, _temp;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var DarthOrgStore = (_class = (_temp = /*#__PURE__*/function () {
  function DarthOrgStore() {
    (0, _classCallCheck2["default"])(this, DarthOrgStore);
    (0, _initializerDefineProperty2["default"])(this, "treeData", _descriptor, this);
    (0, _initializerDefineProperty2["default"])(this, "selectData", _descriptor2, this);
    (0, _initializerDefineProperty2["default"])(this, "getTree", _descriptor3, this);
    (0, _initializerDefineProperty2["default"])(this, "selectTreeOrgDetail", _descriptor4, this);
    (0, _initializerDefineProperty2["default"])(this, "addTree", _descriptor5, this);
    (0, _initializerDefineProperty2["default"])(this, "deleteTree", _descriptor6, this);
    (0, _initializerDefineProperty2["default"])(this, "editTree", _descriptor7, this);
  }

  (0, _createClass2["default"])(DarthOrgStore, [{
    key: "__reactstandin__regenerateByEval",
    // @ts-ignore
    value: function __reactstandin__regenerateByEval(key, code) {
      // @ts-ignore
      this[key] = eval(code);
    }
  }]);
  return DarthOrgStore;
}(), _temp), (_descriptor = (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "treeData", [_mobx.observable], {
  configurable: true,
  enumerable: true,
  writable: true,
  initializer: function initializer() {
    return [];
  }
}), _descriptor2 = (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "selectData", [_mobx.observable], {
  configurable: true,
  enumerable: true,
  writable: true,
  initializer: function initializer() {
    return {
      orgName: '',
      parentOrg: ''
    };
  }
}), _descriptor3 = (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "getTree", [_mobx.action], {
  configurable: true,
  enumerable: true,
  writable: true,
  initializer: function initializer() {
    var _this = this;

    return function (data) {
      var _this2 = this;

      (0, _newArrowCheck2["default"])(this, _this);
      this.treeData = data.map(function (item) {
        (0, _newArrowCheck2["default"])(this, _this2);
        return _objectSpread(_objectSpread({}, item), {}, {
          "default": true
        });
      }.bind(this));
    }.bind(this);
  }
}), _descriptor4 = (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "selectTreeOrgDetail", [_mobx.action], {
  configurable: true,
  enumerable: true,
  writable: true,
  initializer: function initializer() {
    var _this3 = this;

    return function (data) {
      (0, _newArrowCheck2["default"])(this, _this3);
      this.selectData = data;
    }.bind(this);
  }
}), _descriptor5 = (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "addTree", [_mobx.action], {
  configurable: true,
  enumerable: true,
  writable: true,
  initializer: function initializer() {
    var _this4 = this;

    return function (data) {
      var _this5 = this;

      (0, _newArrowCheck2["default"])(this, _this4);
      this.treeData = this.treeData.map(function (item) {
        (0, _newArrowCheck2["default"])(this, _this5);

        if (item.children) {
          return _objectSpread(_objectSpread({}, item), {}, {
            children: item.children.concat(data)
          });
        } else {
          return _objectSpread(_objectSpread({}, item), {}, {
            children: [data]
          });
        }
      }.bind(this));
    }.bind(this);
  }
}), _descriptor6 = (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "deleteTree", [_mobx.action], {
  configurable: true,
  enumerable: true,
  writable: true,
  initializer: function initializer() {
    var _this6 = this;

    return function (id) {
      var _this7 = this;

      (0, _newArrowCheck2["default"])(this, _this6);
      this.treeData = this.treeData.filter(function (item) {
        var _this8 = this;

        (0, _newArrowCheck2["default"])(this, _this7);

        if (item.orgaId !== id) {
          if (item.children && item.children.length > 0) {
            item.children = item.children.filter(function (item) {
              (0, _newArrowCheck2["default"])(this, _this8);
              return item.orgaId !== id;
            }.bind(this));
          }

          return item;
        }
      }.bind(this));
    }.bind(this);
  }
}), _descriptor7 = (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "editTree", [_mobx.action], {
  configurable: true,
  enumerable: true,
  writable: true,
  initializer: function initializer() {
    var _this9 = this;

    return function (data) {
      var _this10 = this;

      (0, _newArrowCheck2["default"])(this, _this9);
      this.treeData = this.treeData.map(function (item) {
        var _this11 = this;

        (0, _newArrowCheck2["default"])(this, _this10);

        if (item.orgaId === data.orgaId) {
          return _objectSpread(_objectSpread({}, item), data);
        } else {
          item.children = item.children.map(function (item) {
            (0, _newArrowCheck2["default"])(this, _this11);

            if (item.orgaId === data.orgaId) {
              return _objectSpread(_objectSpread({}, item), data);
            } else {
              return item;
            }
          }.bind(this));
          return item;
        }
      }.bind(this));
    }.bind(this);
  }
})), _class);
exports.DarthOrgStore = DarthOrgStore;
var DARTH_ORG_STORE = 'darthOrgStore';
exports.DARTH_ORG_STORE = DARTH_ORG_STORE;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(DarthOrgStore, "DarthOrgStore", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/modules/org/store/index.js");
  reactHotLoader.register(DARTH_ORG_STORE, "DARTH_ORG_STORE", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/modules/org/store/index.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
