// darth-orga-ui/publish/lib/darthOrg/components/OrgTree/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

require("antd/es/button/style/css");

var _button = _interopRequireDefault(require("antd/es/button"));

require("antd/es/input/style/css");

var _input = _interopRequireDefault(require("antd/es/input"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

var _react = _interopRequireWildcard(require("react"));

var _mobxReact = require("mobx-react");

var _classes = require("../../../../utils/classes");

var _DarthTree = _interopRequireDefault(require("../../../../common/DarthTree"));

var _addOrg = _interopRequireDefault(require("../addOrg"));

require("./index.scss");

var _this = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var sc = (0, _classes.scopedClassMaker)('orga-tree');

var OrgTree = __signature__(function OrgTree(props) {
  var _this2 = this;

  (0, _newArrowCheck2["default"])(this, _this);
  var classNameTree = props.classNameTree,
      treeData = props.treeData,
      onSelect = props.onSelect,
      darthOrgStore = props.darthOrgStore;
  var deleteTree = darthOrgStore.deleteTree;

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      visible = _useState2[0],
      setVisible = _useState2[1];

  var _useState3 = (0, _react.useState)({}),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      editData = _useState4[0],
      setEditData = _useState4[1];

  var _useState5 = (0, _react.useState)(false),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      isEdit = _useState6[0],
      setEdit = _useState6[1];

  var onAddOrg = function onAddOrg() {
    (0, _newArrowCheck2["default"])(this, _this2);
    setVisible(true);
  }.bind(this);

  var cancelAddOrg = function cancelAddOrg() {
    (0, _newArrowCheck2["default"])(this, _this2);
    setVisible(false);
    setEditData({});
    setEdit(false);
  }.bind(this);

  var onDeleteTree = function onDeleteTree(data) {
    (0, _newArrowCheck2["default"])(this, _this2);
    deleteTree(data.id);
  }.bind(this);

  var onEditTree = function onEditTree(data) {
    (0, _newArrowCheck2["default"])(this, _this2);
    setEditData(data);
    setEdit(true);
    setVisible(true);
  }.bind(this);

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: sc('wrap', {
      extra: [classNameTree].join(' ')
    })
  }, /*#__PURE__*/_react["default"].createElement(_input["default"], {
    placeholder: '搜索组织名称'
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: sc('wrap-tree')
  }, /*#__PURE__*/_react["default"].createElement(_DarthTree["default"], {
    data: treeData,
    onSelect: onSelect,
    defaultExpandedKeys: ['key1', 'key1-2'],
    onDelete: onDeleteTree,
    onEdit: onEditTree
  })), /*#__PURE__*/_react["default"].createElement(_button["default"], {
    type: "primary",
    onClick: onAddOrg
  }, "+\u6DFB\u52A0\u7EC4\u7EC7"), /*#__PURE__*/_react["default"].createElement(_addOrg["default"], (0, _extends2["default"])({}, props, {
    visible: visible,
    cancelAddOrg: cancelAddOrg,
    parentTree: treeData,
    editData: editData,
    isEdit: isEdit
  })));
}, "useState{[visible, setVisible](false)}\nuseState{[editData, setEditData]({})}\nuseState{[isEdit, setEdit](false)}").bind(void 0);

var _default = (0, _mobxReact.observer)(OrgTree);

var _default2 = _default;
exports["default"] = _default2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(sc, "sc", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthOrg/components/OrgTree/index.js");
  reactHotLoader.register(OrgTree, "OrgTree", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthOrg/components/OrgTree/index.js");
  reactHotLoader.register(_default, "default", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthOrg/components/OrgTree/index.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
