// darth-orga-ui/publish/common/OrgMenu/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

require("antd/es/menu/style/css");

var _menu = _interopRequireDefault(require("antd/es/menu"));

var _react = _interopRequireDefault(require("react"));

var _mobxReact = require("mobx-react");

require("./index.scss");

var _utils = require("../../utils");

var _this = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var SubMenu = _menu["default"].SubMenu;
var sc = (0, _utils.scopedClassMaker)('orga-header');

var MenuList = function MenuList(props) {
  var _this2 = this;

  (0, _newArrowCheck2["default"])(this, _this);
  var className = props.className,
      onSelectMenu = props.onSelectMenu;

  var _loop = function loop() {
    var _this3 = this;

    var data = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
    (0, _newArrowCheck2["default"])(this, _this2);
    return data.map(function (item, index) {
      (0, _newArrowCheck2["default"])(this, _this3);

      if (item.children && item.children.length) {
        return /*#__PURE__*/_react["default"].createElement(SubMenu, {
          key: "".concat(item.id),
          title: item.title,
          icon: item.icon
        }, _loop(item.children));
      }

      return /*#__PURE__*/_react["default"].createElement(_menu["default"].Item, {
        icon: item.icon,
        key: "".concat(item.id)
      }, item.title);
    }.bind(this));
  }.bind(this);

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: sc('menu', {
      extra: [className].join(' ')
    })
  }, /*#__PURE__*/_react["default"].createElement(_menu["default"], {
    theme: "light",
    mode: 'inline',
    defaultSelectedKeys: props.defaultSelectedKeys,
    defaultOpenKeys: props.defaultOpenKeys,
    onSelect: onSelectMenu
  }, _loop(props.data)));
}.bind(void 0);

var _default = (0, _mobxReact.observer)(MenuList);

var _default2 = _default;
exports["default"] = _default2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(SubMenu, "SubMenu", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/common/OrgMenu/index.js");
  reactHotLoader.register(sc, "sc", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/common/OrgMenu/index.js");
  reactHotLoader.register(MenuList, "MenuList", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/common/OrgMenu/index.js");
  reactHotLoader.register(_default, "default", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/common/OrgMenu/index.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
