// darth-orga/publish/modules/org/components/deltails-org.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

require("antd/es/table/style/css");

var _table = _interopRequireDefault(require("antd/es/table"));

require("antd/es/button/style/css");

var _button = _interopRequireDefault(require("antd/es/button"));

require("antd/es/row/style/css");

var _row = _interopRequireDefault(require("antd/es/row"));

require("antd/es/col/style/css");

var _col = _interopRequireDefault(require("antd/es/col"));

require("antd/es/input/style/css");

var _input = _interopRequireDefault(require("antd/es/input"));

require("antd/es/space/style/css");

var _space = _interopRequireDefault(require("antd/es/space"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

require("antd/es/tabs/style/css");

var _tabs = _interopRequireDefault(require("antd/es/tabs"));

var _react = _interopRequireWildcard(require("react"));

require("./deltails-org.scss");

var _utils = require("../../../common/utils");

var _userSelect = _interopRequireDefault(require("./userSelect"));

var _addUser = _interopRequireDefault(require("../../user/components/add-user"));

var _this = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var TabPane = _tabs["default"].TabPane;

var OraDetailOrgName = __signature__(function OraDetailOrgName(props) {
  var _this2 = this;

  (0, _newArrowCheck2["default"])(this, _this);
  var data = props.data;

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      visible = _useState2[0],
      setVisible = _useState2[1];

  var columns = [{
    title: '用户名',
    dataIndex: 'name',
    key: 'name'
  }, {
    title: '帐号',
    dataIndex: 'account',
    key: 'account'
  }, {
    title: '所属机构',
    dataIndex: 'org',
    key: 'org'
  }, {
    title: '类别',
    dataIndex: 'userType',
    key: 'userType',
    render: function render(text) {
      (0, _newArrowCheck2["default"])(this, _this2);

      switch (text) {
        case 2:
          return '第三方';

        case 1:
          return '内部';

        default:
          return '--';
      }
    }.bind(this)
  }, {
    title: '操作',
    key: 'action',
    render: function render(text, record) {
      var _this3 = this;

      (0, _newArrowCheck2["default"])(this, _this2);
      return /*#__PURE__*/_react["default"].createElement(_space["default"], {
        size: "middle"
      }, /*#__PURE__*/_react["default"].createElement("a", {
        onClick: function () {
          (0, _newArrowCheck2["default"])(this, _this3);
          return deleteUser(record.id);
        }.bind(this)
      }, "\u5220\u9664"));
    }.bind(this)
  }];

  var deleteUser = function deleteUser(id) {
    (0, _newArrowCheck2["default"])(this, _this2);
  }.bind(this);

  var datas = [{
    key: '1',
    name: 'John Brown',
    account: 32,
    org: 'New York No. 1 Lake Park',
    userType: 2
  }, {
    key: '2',
    name: 'Jim Green',
    account: 42,
    org: 'London No. 1 Lake Park',
    userType: 1
  }, {
    key: '3',
    name: 'Joe Black',
    account: 32,
    org: 'Sidney No. 1 Lake Park',
    userType: 2
  }];

  var openUserList = function openUserList() {
    (0, _newArrowCheck2["default"])(this, _this2);
    setVisible(true);
  }.bind(this);

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: 'org-detail'
  }, /*#__PURE__*/_react["default"].createElement(_row["default"], null, /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 8
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: 'org-detail-item'
  }, /*#__PURE__*/_react["default"].createElement("label", null, "\u673A\u6784\u540D\u79F0:"), /*#__PURE__*/_react["default"].createElement(_input["default"], {
    value: data.orgName,
    disabled: true
  })))), /*#__PURE__*/_react["default"].createElement(_row["default"], null, /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 8
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: 'org-detail-item'
  }, /*#__PURE__*/_react["default"].createElement("label", null, "\u4E0A\u7EA7\u673A\u6784:"), /*#__PURE__*/_react["default"].createElement(_input["default"], {
    value: data.parentOrg,
    disabled: true
  })))), /*#__PURE__*/_react["default"].createElement(_row["default"], null, /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 24
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "card-container"
  }, /*#__PURE__*/_react["default"].createElement(_tabs["default"], {
    type: "card"
  }, /*#__PURE__*/_react["default"].createElement(TabPane, {
    tab: "\u7528\u6237",
    key: "1"
  }, /*#__PURE__*/_react["default"].createElement(_row["default"], null, /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 6
  }, /*#__PURE__*/_react["default"].createElement(_input["default"], {
    placeholder: "\u641C\u7D22\u7528\u6237\u540D\u79F0"
  })), /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 2,
    offset: 16
  }, /*#__PURE__*/_react["default"].createElement(_button["default"], {
    type: "primary",
    onClick: openUserList
  }, "+\u6DFB\u52A0\u7528\u6237"))), /*#__PURE__*/_react["default"].createElement(_row["default"], null, /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 24
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: 'org-detail-table'
  }, /*#__PURE__*/_react["default"].createElement(_table["default"], {
    bordered: true,
    columns: columns,
    dataSource: datas
  }))))))))), /*#__PURE__*/_react["default"].createElement(_userSelect["default"], (0, _extends2["default"])({}, props, {
    visible: visible,
    onCancel: function () {
      (0, _newArrowCheck2["default"])(this, _this2);
      return setVisible(false);
    }.bind(this)
  })));
}, "useState{[visible, setVisible](false)}").bind(void 0);

var _default = OraDetailOrgName;
var _default2 = _default;
exports["default"] = _default2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(TabPane, "TabPane", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/modules/org/components/deltails-org.js");
  reactHotLoader.register(OraDetailOrgName, "OraDetailOrgName", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/modules/org/components/deltails-org.js");
  reactHotLoader.register(_default, "default", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/modules/org/components/deltails-org.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
