// darth-orga/publish/modules/org/components/userSelect.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("antd/es/modal/style/css");

var _modal = _interopRequireDefault(require("antd/es/modal"));

require("antd/es/table/style/css");

var _table = _interopRequireDefault(require("antd/es/table"));

require("antd/es/row/style/css");

var _row = _interopRequireDefault(require("antd/es/row"));

require("antd/es/col/style/css");

var _col = _interopRequireDefault(require("antd/es/col"));

require("antd/es/button/style/css");

var _button = _interopRequireDefault(require("antd/es/button"));

require("antd/es/form/style/css");

var _form = _interopRequireDefault(require("antd/es/form"));

require("antd/es/input/style/css");

var _input = _interopRequireDefault(require("antd/es/input"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

var _react = _interopRequireWildcard(require("react"));

var _utils = require("../../../common/utils");

var _this = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var UserSelect = __signature__(function UserSelect(props) {
  var _this2 = this;

  (0, _newArrowCheck2["default"])(this, _this);
  var visible = props.visible,
      onCancel = props.onCancel;

  var _useState = (0, _react.useState)([]),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      selectedRowKeys = _useState2[0],
      setSelectedRowKeys = _useState2[1];

  var _useState3 = (0, _react.useState)([]),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      users = _useState4[0],
      setUsers = _useState4[1];

  (0, _react.useEffect)(function () {
    var _this3 = this;

    (0, _newArrowCheck2["default"])(this, _this2);
    (0, _utils.baseFetch)('post', 'api/user/findAllUser').then(function (res) {
      (0, _newArrowCheck2["default"])(this, _this3);

      if (!res.code) {
        setUsers(res.data);
      }
    }.bind(this));
  }.bind(this), []);

  var handleOk = function handleOk() {
    (0, _newArrowCheck2["default"])(this, _this2);
  }.bind(this);

  var handleCancel = function handleCancel() {
    (0, _newArrowCheck2["default"])(this, _this2);
    onCancel();
  }.bind(this);

  var onSearch = function onSearch(values) {
    (0, _newArrowCheck2["default"])(this, _this2);
  }.bind(this);

  var columns = [{
    title: '姓名',
    dataIndex: 'name',
    key: 'name',
    width: '20%',
    ellipsis: true
  }, {
    title: '手机号',
    dataIndex: 'phone',
    key: 'phone',
    width: '20%',
    ellipsis: true
  }, {
    title: '邮箱',
    dataIndex: 'email',
    key: 'email',
    width: '20%',
    ellipsis: true
  }, {
    title: '部门',
    dataIndex: 'org',
    key: 'org',
    width: '20%',
    ellipsis: true
  }, {
    title: '类别',
    dataIndex: 'userType',
    key: 'userType',
    width: '10%',
    ellipsis: true,
    render: function render(text) {
      (0, _newArrowCheck2["default"])(this, _this2);

      switch (text) {
        case 2:
          return '第三方';

        case 1:
          return '内部';

        default:
          return '--';
      }
    }.bind(this)
  }];

  var onSelectChange = function onSelectChange(selectedRowKeys) {
    (0, _newArrowCheck2["default"])(this, _this2);
    console.log('selectedRowKeys changed: ', selectedRowKeys);
    setSelectedRowKeys(selectedRowKeys);
  }.bind(this);

  var rowSelection = {
    selectedRowKeys: selectedRowKeys,
    onChange: onSelectChange
  };
  return /*#__PURE__*/_react["default"].createElement(_modal["default"], {
    title: "\u9009\u62E9\u7528\u6237",
    visible: visible,
    onOk: handleOk,
    onCancel: handleCancel,
    closable: false,
    width: 800
  }, /*#__PURE__*/_react["default"].createElement(_row["default"], null, /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 24
  }, /*#__PURE__*/_react["default"].createElement(_form["default"], {
    layout: "inline",
    onFinish: onSearch
  }, /*#__PURE__*/_react["default"].createElement(_form["default"].Item, {
    name: "name",
    label: "\u7528\u6237\u540D"
  }, /*#__PURE__*/_react["default"].createElement(_input["default"], null)), /*#__PURE__*/_react["default"].createElement(_form["default"].Item, null, /*#__PURE__*/_react["default"].createElement(_button["default"], {
    type: "primary",
    htmlType: "submit"
  }, "\u641C\u7D22"))))), /*#__PURE__*/_react["default"].createElement(_row["default"], null, /*#__PURE__*/_react["default"].createElement(_col["default"], {
    span: 24
  }, /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      width: '100%'
    }
  }, /*#__PURE__*/_react["default"].createElement(_table["default"], {
    columns: columns,
    dataSource: users,
    rowSelection: rowSelection,
    rowKey: function (record) {
      (0, _newArrowCheck2["default"])(this, _this2);
      return record.id;
    }.bind(this)
  })))));
}, "useState{[selectedRowKeys, setSelectedRowKeys]([])}\nuseState{[users, setUsers]([])}\nuseEffect{}").bind(void 0);

var _default = UserSelect;
var _default2 = _default;
exports["default"] = _default2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(UserSelect, "UserSelect", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/modules/org/components/userSelect.js");
  reactHotLoader.register(_default, "default", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/modules/org/components/userSelect.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
