// darth-orga-ui/publish/lib/darthOrg/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

var _react = _interopRequireWildcard(require("react"));

var _mobxReact = require("mobx-react");

var _OrgTree = _interopRequireDefault(require("./components/OrgTree"));

var _breadCrumb = _interopRequireDefault(require("./components/breadCrumb"));

var _deltailsOrgName = _interopRequireDefault(require("./components/deltailsOrgName"));

var _utils = require("../../utils");

require("./index.scss");

var _this = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var DarthOrgContent = __signature__(function DarthOrgContent(props) {
  var _this2 = this;

  (0, _newArrowCheck2["default"])(this, _this);
  var darthOrgStore = props.darthOrgStore,
      selectMenu = props.selectMenu;
  var treeData = darthOrgStore.treeData,
      selectData = darthOrgStore.selectData,
      getTree = darthOrgStore.getTree,
      selectTreeOrgDetail = darthOrgStore.selectTreeOrgDetail;
  var crumbRoutes = [{
    path: 'index',
    breadcrumbName: 'home'
  }, {
    path: 'first',
    breadcrumbName: 'first'
  }, {
    path: 'second',
    breadcrumbName: 'second'
  }]; // 初始化 组织树的数据

  (0, _react.useEffect)(function () {
    var _this3 = this;

    (0, _newArrowCheck2["default"])(this, _this2);
    (0, _utils.baseFetch)('get', 'mock/tree').then(function (res) {
      (0, _newArrowCheck2["default"])(this, _this3);

      if (!res.error) {
        getTree(res.data);
      }
    }.bind(this)); // baseFetch('post', 'api/orga/findAllOrga',).then(res=>{
    //     if (!res.code) {
    //         getTree(res.data)
    //     }
    // })
  }.bind(this), []); // 选中树的数据

  var onSelectTree = function onSelectTree(data) {
    var _this4 = this;

    (0, _newArrowCheck2["default"])(this, _this2);
    (0, _utils.baseFetch)('get', 'mock/org').then(function (res) {
      (0, _newArrowCheck2["default"])(this, _this4);
      selectTreeOrgDetail({
        orgName: res.data.org,
        parentOrg: res.data.parent
      });
    }.bind(this))["catch"](function (err) {
      (0, _newArrowCheck2["default"])(this, _this4);
      console.log(err);
    }.bind(this));
  }.bind(this);

  var onSelectMenu = function onSelectMenu(e) {
    (0, _newArrowCheck2["default"])(this, _this2);

    if (typeof selectMenu === 'function') {
      selectMenu(e);
    } else {
      console.error('selectMenu 必须是函数方法');
    }
  }.bind(this);

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "org-parent"
  }, /*#__PURE__*/_react["default"].createElement(_OrgTree["default"], (0, _extends2["default"])({
    treeData: treeData,
    onSelect: onSelectTree
  }, props)), /*#__PURE__*/_react["default"].createElement("div", {
    className: "org-parent-detail"
  }, /*#__PURE__*/_react["default"].createElement(_breadCrumb["default"], {
    routes: crumbRoutes
  }), /*#__PURE__*/_react["default"].createElement(_deltailsOrgName["default"], {
    data: selectData
  }), "\u7528\u6237\u8868"));
}, "useEffect{}").bind(void 0);

var _default = (0, _mobxReact.inject)('darthOrgStore')((0, _mobxReact.observer)(DarthOrgContent));

var _default2 = _default;
exports["default"] = _default2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(DarthOrgContent, "DarthOrgContent", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthOrg/index.js");
  reactHotLoader.register(_default, "default", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthOrg/index.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
