// darth-orga-ui/publish/lib/darthOrg/components/addOrg/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("antd/es/modal/style/css");

var _modal = _interopRequireDefault(require("antd/es/modal"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

require("antd/es/input/style/css");

var _input = _interopRequireDefault(require("antd/es/input"));

require("antd/es/select/style/css");

var _select = _interopRequireDefault(require("antd/es/select"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

require("antd/es/form/style/css");

var _form = _interopRequireDefault(require("antd/es/form"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _newArrowCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/newArrowCheck"));

var _react = _interopRequireWildcard(require("react"));

require("./index.scss");

var _utils = require("../../../../utils");

var _this = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var layout = {
  labelCol: {
    span: 6
  },
  wrapperCol: {
    span: 18
  }
};

var AddOrg = __signature__(function AddOrg(props) {
  var _this2 = this;

  (0, _newArrowCheck2["default"])(this, _this);
  var visible = props.visible,
      cancelAddOrg = props.cancelAddOrg,
      parentTree = props.parentTree,
      darthOrgStore = props.darthOrgStore,
      _props$editData = props.editData,
      editData = _props$editData === void 0 ? {} : _props$editData,
      _props$isEdit = props.isEdit,
      isEdit = _props$isEdit === void 0 ? false : _props$isEdit;
  var addTree = darthOrgStore.addTree,
      editTree = darthOrgStore.editTree;

  var _Form$useForm = _form["default"].useForm(),
      _Form$useForm2 = (0, _slicedToArray2["default"])(_Form$useForm, 1),
      form = _Form$useForm2[0];

  (0, _react.useEffect)(function () {
    (0, _newArrowCheck2["default"])(this, _this2);

    if (parentTree.length && isEdit) {
      form.setFieldsValue({
        parentName: parentTree[0].id
      });
      form.setFieldsValue({
        org: editData.title
      });
    }
  }.bind(this), [isEdit, editData, parentTree]);

  var handleOk = function handleOk() {
    (0, _newArrowCheck2["default"])(this, _this2);
    form.submit();
  }.bind(this);

  var handleCancel = function handleCancel() {
    (0, _newArrowCheck2["default"])(this, _this2);
    form.resetFields();
    cancelAddOrg();
  }.bind(this);

  var onFinish = function onFinish(values) {
    var _this3 = this;

    (0, _newArrowCheck2["default"])(this, _this2);

    if (isEdit) {
      (0, _utils.baseFetch)('post', 'mock/editTree', _objectSpread(_objectSpread({}, values), {}, {
        id: editData.id
      })).then(function (res) {
        (0, _newArrowCheck2["default"])(this, _this3);

        if (!res.error) {
          editTree(res.data);
          form.resetFields();
          cancelAddOrg();
        }
      }.bind(this));
    } else {
      (0, _utils.baseFetch)('post', 'mock/addTree', values).then(function (res) {
        (0, _newArrowCheck2["default"])(this, _this3);

        if (!res.error) {
          addTree(res.data);
          form.resetFields();
          cancelAddOrg();
        }
      }.bind(this));
    }
  }.bind(this);

  var renderSelectOption = function renderSelectOption() {
    var _this4 = this;

    (0, _newArrowCheck2["default"])(this, _this2);
    return parentTree.map(function (item) {
      (0, _newArrowCheck2["default"])(this, _this4);
      return /*#__PURE__*/_react["default"].createElement(_select["default"].Option, {
        value: item.id,
        key: item.id
      }, item.title);
    }.bind(this));
  }.bind(this); // const formValues = isEdit ? {
  //     ['parentName']:parentTree[0].id,
  //     ['org']: editData.title,
  // } : form.resetFields()
  // console.log(formValues)


  return /*#__PURE__*/_react["default"].createElement(_modal["default"], {
    title: "\u6DFB\u52A0\u7EC4\u7EC7",
    visible: visible,
    onOk: handleOk,
    onCancel: handleCancel,
    closable: false
  }, /*#__PURE__*/_react["default"].createElement(_form["default"], (0, _extends2["default"])({}, layout, {
    form: form,
    onFinish: onFinish // initialValues={formValues}

  }), /*#__PURE__*/_react["default"].createElement(_form["default"].Item, {
    label: "\u673A\u6784\u540D\u79F0",
    name: "org",
    rules: [{
      required: true,
      message: '用户名不能包含非法字符，如&,%，&，#……等'
    }]
  }, /*#__PURE__*/_react["default"].createElement(_input["default"], null)), /*#__PURE__*/_react["default"].createElement(_form["default"].Item, {
    label: "\u4E0A\u7EA7\u673A\u6784",
    name: "parentName",
    rules: [{
      required: true,
      message: '上级机构必填!'
    }]
  }, /*#__PURE__*/_react["default"].createElement(_select["default"], null, renderSelectOption()))));
}, "useForm{[form]}\nuseEffect{}").bind(void 0);

var _default = AddOrg;
var _default2 = _default;
exports["default"] = _default2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(layout, "layout", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthOrg/components/addOrg/index.js");
  reactHotLoader.register(AddOrg, "AddOrg", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthOrg/components/addOrg/index.js");
  reactHotLoader.register(_default, "default", "/Users/mahai/Desktop/darth-platform-ui/packages/darth-orga-ui/src/components/darthOrg/components/addOrg/index.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
